tests/test_subscription.py

```diff
@@ -308,3 +308,32 @@ def test_old_django_connections_closed_when_middleware_is_used(
 
         assert res == 123
         assert mock_close_old_connections.call_count == 2
+
+
+class TestDecorator:
+    def test_returns_susbscription_when_function_valid(self):
+        subscription = sub(topic="topic", prefix="rele")(lambda data: None)
+        assert isinstance(subscription, Subscription)
+
+    def test_raises_error_when_function_signature_is_not_valid(self):
+        with pytest.raises(RuntimeError):
+            sub(topic="topic", prefix="rele")(lambda: None)
+
+    def test_logs_warning_when_function_not_in_subs_module(self, caplog):
+        sub(topic="topic", prefix="rele")(lambda data: None)
+        assert (
+            "Subscription function tests.test_subscription.<lambda> is outside a subs "
+        )
+        "module that will not be discovered." in caplog.text
```

Comment:
Is there something wrong with the formatting here?

Reply:
Black lied to me! I'll change it